CMakeLists.txt

```diff
@@ -10,7 +10,23 @@ project(SQLiteCpp VERSION "2.99")
 
 # SQLiteC++ 3.x now requires C++11 compiler
 set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+option(USE_LATEST_CXX_STANDARD_VERSION "Makes CMake to probe for the latest CXX version. May give efficiency improvements. Disables mandating minimal standard version: CMake has design flaws that prevents from adding a minimal version constraint while opportunistically upgrading: CMAKE_CXX_STANDARD_REQUIRED should have been an int for that, not a boolean" OFF)
+if(USE_LATEST_CXX_STANDARD_VERSION)
+    if(${CMAKE_VERSION} VERSION_GREATER "3.12") 
+        set(CMAKE_CXX_STANDARD 20)
+    else()
+        if(${CMAKE_VERSION} VERSION_GREATER "3.8") 
+            set(CMAKE_CXX_STANDARD 17)
+        else()
+            set(CMAKE_CXX_STANDARD 14)
+        endif()
+    endif()
+else()
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
could you also move the line set(CMAKE_CXX_STANDARD 11) here so it will make more sense (to have it all in the same if/else)